plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -75,22 +68,35 @@ func (adx *AzureDataExplorer) Connect() error {
 		Authorizer: authorizer,
 	}
 	client, err := kusto.New(adx.Endpoint, authorization)
-
+	adx.Log.Debug("Connect : Client initialized successfully")
 	if err != nil {
 		return err
 	}
 	adx.client = client
-	adx.ingesters = make(map[string]localIngestor)
-	adx.createIngestor = createRealIngestor
-
+	adx.ingestors = make(map[string]ingest.Ingestor)
 	return nil
 }
 
+// Clean up and close the ingestor
 func (adx *AzureDataExplorer) Close() error {
+	var err error
+	for _, v := range adx.ingestors {
+		err = v.Close()
```

Comment:
Shouldn't you check this error? In your current code, you will only check the code of the last `adx.ingestors`...